cunumeric/module.py

```diff
@@ -1842,23 +1913,23 @@ def array_split(a, indices, axis=0, equal=False):
                 )
                 first_idx = (len_subarr + 1) * res
         split_pts.extend(range(first_idx, end_idx + 1, len_subarr))
-    elif (
-        (dtype == np.ndarray and indices.dtype == int)
-        or dtype == list
-        or dtype == tuple
+
+    elif isinstance(indices, (list, tuple)) or (
+        isinstance(indices, (ndarray, np.ndarray)) and indices.dtype == int
```

Comment:
@magnatelee your comment here seems to have disappeared. In any case, updated in https://github.com/nv-legate/cunumeric/pull/413/commits/88b54dd36af1745285d6dd374dcf9ecb1ba0de34 I think `Sequence[int]` is a slight fib since we really only handle lists and tuples but I think it's ok